src/color.rs

```diff
@@ -106,36 +158,265 @@ impl ToCss for RGBA {
     where
         W: fmt::Write,
     {
-        let serialize_alpha = self.alpha != 255;
+        let has_alpha = self.alpha != 255;
 
-        dest.write_str(if serialize_alpha { "rgba(" } else { "rgb(" })?;
+        dest.write_str(if has_alpha { "rgba(" } else { "rgb(" })?;
         self.red.to_css(dest)?;
         dest.write_str(", ")?;
         self.green.to_css(dest)?;
         dest.write_str(", ")?;
         self.blue.to_css(dest)?;
-        if serialize_alpha {
-            dest.write_str(", ")?;
 
-            // Try first with two decimal places, then with three.
-            let mut rounded_alpha = (self.alpha_f32() * 100.).round() / 100.;
-            if clamp_unit_f32(rounded_alpha) != self.alpha {
-                rounded_alpha = (self.alpha_f32() * 1000.).round() / 1000.;
+        serialize_alpha(dest, self.alpha, true)?;
+
+        dest.write_char(')')
+    }
+}
+
+// NOTE: LAB and OKLAB is not declared inside the [impl_lab_like] macro,
+// because it causes cbindgen to ignore them.
+
+/// Color specified by lightness, a- and b-axis components.
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[repr(C)]
+pub struct LAB {
+    /// The lightness component.
+    pub lightness: f32,
+    /// The a-axis component.
+    pub a: f32,
+    /// The b-axis component.
+    pub b: f32,
+    /// The alpha component.
+    pub alpha: u8,
```

Comment:
I think maybe storing the alpha as an f32 might be better long term (avoids having different computed / animated representations). Wdyt?

Reply:
That would make things easier yes.  I think I'm going to do it with `RGBA.alpha` as well.  Right now the size of the struct doesn't matter (we'll see if this is an issue) and dealing with alpha values the same would also makes things much simpler.  Can you think of issues with this?

Reply:
The main issue is that it'd slow down conversion to `nscolor` (the Gecko representation of a computed rgba value which is ~everywhere). But might not be a huge issue.

Reply:
We'll have to convert to rgba (nscolor) anyway from lab/lch/etc. If this becomes a problem we can optimize.